tests/SafeFloatTest.php

```diff
@@ -0,0 +1,126 @@
+    }
+
+    /** @dataProvider impossibleFloatValues */
+    function test_throwing_InvalidArgumentException_on_value_that_cannot_be_float($notFloat)
+    {
+        $this->expectException(InvalidArgumentException::class);
+        SafeAssocArray::from(['value' => $notFloat])->float('value');
+    }
+
+    /** @dataProvider impossibleFloatValues */
+    function test_throwing_InvalidArgumentException_on_value_that_cannot_be_float_nullable($notFloat)
+    {
+        $this->expectException(InvalidArgumentException::class);
+        SafeAssocArray::from(['value' => $notFloat])->floatNullable('value');
+    }
+
+    function test_throwing_InvalidArgumentException_on_null()
+    {
+        $this->expectException(InvalidArgumentException::class);
+        SafeAssocArray::from(['value' => null])->float('value');
+    }
+
+    /** @dataProvider impossibleFloatValues */
+    function test_returning_default_when_value_is_not_numeric($notFloat)
+    {
+        $safe = SafeAssocArray::from(['value' => $notFloat]);
+        $float = (float)(random_int(100, 200) / random_int(1, 100));
+
+        self::assertSame($float, $safe->floatOrDefault('value', $float));
+    }
+
+    function test_casts_array_of_values_that_can_be_float()
+    {
+        self::assertEquals(
+            [123.0, 1.23, 1.0, 0.0, 123.99],
+            SafeAssocArray::from(
+                [
+                    'floats' => [
+                        123,
+                        1.23,
+                        true,
+                        false,
+                        new StringObject('123.99'),
+                    ],
+                ]
+            )->floats('floats')
+        );
+    }
+
+    function test_throws_InvalidArgumentException_when_cannot_cast_array_of_floats()
+    {
+        $this->expectException(InvalidArgumentException::class);
+        SafeAssocArray::from(['notOnlyFloats' => ['123', new stdClass()]])->floats('notOnlyFloats');
+    }
+
+    function test_filtered_casting_only_floats_from_array_of_mixed_values()
+    {
+        self::assertEquals(
+            [123.99, 456.99],
+            SafeAssocArray::from(
+                [
+                    'floats' => [
+                        '123.99',
+                        ['array'],
+                        new StringObject('456.99'),
+                        null,
+                    ],
+                ]
+            )->floatsFiltered('floats')
+        );
+    }
+
+    function test_casting_array_of_mixed_values_with_provided_default()
+    {
+        self::assertEquals(
+            [123.99, 42.42, 456.99, 42.42],
+            SafeAssocArray::from(
+                [
+                    'floats' => [
+                        '123.99',
+                        ['array'],
+                        new StringObject('456.99'),
+                        null,
+                    ],
+                ]
+            )->floatsForced('floats', 42.42)
```

Comment:
brakuje testu na `floatsForced` bez `default`

Reply:
a, tam jest `0` w default, więc chyba nie trzeba osobno testować

Reply:
jednak dodałem przypadek na to